### PR TITLE
Add 4xx Http Status Alerting

### PR DESCRIPTION
**Pull request by saquino0827**
# Describe

- Added an alert for http status code 4xx.
- Made the alert stateful with a frequency check of 1 minute and a look back period of 5 minutes.
  -  I am looking for opinions on frequency and look back periods.
  - There is a comment in the code where this is set and I will remove that comment when we have a decision on the timings.

## Issue

[1396](https://github.com/CDCgov/trusted-intermediary/issues/1396)

## Checklist

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: operations/template/alert.tf
Comment (halprin) [inline operations/template/alert.tf:87]: The reason the tags were enumerated separately was because we only wanted to ignore the automatically added tags from the CDC, not any tags added by us if we ever were to.

File: operations/template/alert.tf
Comment (halprin) [inline operations/template/alert.tf:100]: The real question to me is how many 4XXs in so many minutes do we want to initiate an incident?  Like, if one of us does a test in production, and we accidentally hit the wrong URL that generates a 404, is that something worth generating an incident over?  Or what about if ReportStream calls us, they have accidentally lost their secrets for us, and so they start getting 401s?

So, to answer the question, I'm not sure how big the `window_size` should be, but it is impacted by the `threshold` below too.  I feel `threshold` should be greater than `1` at least.

Comment (halprin) [reply operations/template/alert.tf:133]: Thinking about this some more, perhaps the threshold should be 3 4XXs within an hour?  So `threshold` will be `3`, `frequency` can be `PT1M` (every minute) and `window_size` is `PT1H` (or `60M` if `1H` isn't valid?).  Thoughts?

File: operations/template/alert.tf
Comment (halprin) [inline operations/template/alert.tf:129]: We don't want the entire resource group checked, right?  Since this is in the TI repo, we want this specific to TI, so should this be perhaps `azurerm_linux_web_app.api.id`?

File: operations/template/alert.tf
Comment (saquino0827) [reply operations/template/alert.tf]: See: https://github.com/CDCgov/trusted-intermediary/issues/1396#issuecomment-2429878961

We reverted back to the previous style of `ignore_changes`

File: operations/template/alert.tf
Comment (saquino0827) [reply operations/template/alert.tf:133]: I'd have to check what are valid values but if I can I will set it to 1H.

Comment (saquino0827) [reply operations/template/alert.tf]: This has been updated to the recommended.

File: operations/template/alert.tf
Comment (saquino0827) [reply operations/template/alert.tf]: This is outdated due to https://github.com/CDCgov/trusted-intermediary/issues/1396#issuecomment-2429878961

File: operations/template/alert.tf
Comment (halprin) [reply operations/template/alert.tf:129]: @saquino0827, I think all I'm waiting for is addressing this feedback (or having a conversation around this if you feel this isn't right).  After that, I'm ready to approve.

Comment (halprin) [reply operations/template/alert.tf:129]: Oh, and also probably want to make sure this actually works in some environment.

Comment (saquino0827) [reply operations/template/alert.tf]: @halprin this has been updated. Currently testing in internal and waiting for the alert that was fired to reach resolved status

**Review by halprin — APPROVED** [review]
Looks like it works.  Nice.

**Pull request merged**
